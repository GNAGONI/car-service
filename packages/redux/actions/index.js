export * from './auctions';
export * from './longAuctions';
export * from './dropDowns';
export * from './car';
export * from './location';
export * from './user';
export * from './contactUs';
export * from './toasts';
export * from './quote';
