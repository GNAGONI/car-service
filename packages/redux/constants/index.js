import UserTypes from '../action-types/user';

export const STATUS = {
  IDLE: 'idle',
  RUNNING: 'running',
  READY: 'ready',
  SUCCESS: 'success',
  ERROR: 'error',
};

export const ignoredActionTypes = [
  UserTypes.USER_LOGIN.REQUEST,
  UserTypes.USER_SIGNUP.REQUEST,
  UserTypes.REFRESH_TOKEN.REQUEST,
  UserTypes.USER_LOGOUT.REQUEST,
  UserTypes.FORGOT_PASSWORD.REQUEST,
];

export const DELAY_TIME = 1000;

export const DEFAULT_POSTCODE = 'WC2N 5DU';

export const SUCCESS_BID = 'SUCCESS_BID';

export const SOME_ERROR_OCCURRED = 'There were errors. Please fix them and try again';
export const SUCCESSFULLY_LOGGED_IN = 'You have been successfully logged in';
export const SUCCESSFULLY_SIGNED_UP = 'You have successfully created an account on Car.service';
