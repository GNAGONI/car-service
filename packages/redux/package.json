{
   "name": "@car-service/redux",
   "version": "0.0.1-alpha.0",
   "private": true,
   "scripts": {
      "lint": "eslint . --ext .js --ext .jsx",
      "lint:fix": "yarn lint -- --fix",
      "lint:quiet": "yarn lint -- --quiet",
      "prettier": "prettier './**/*.js' --write --single-quote",
      "prettier:write": "yarn prettier -- --write",
      "prettier:check": "yarn prettier -- --list-different"
   },
   "dependencies": {
      "@car-service/api-client": "^0.0.1-alpha.0",
      "immutability-helper": "^3.0.0",
      "isomorphic-unfetch": "^3.0.0",
      "keymirror": "^0.1.1",
      "moment": "^2.24.0",
      "redux": "^4.0.1",
      "redux-actions": "^2.6.4",
      "redux-devtools-extension": "^2.13.8",
      "redux-saga": "^1.0.0",
      "reselect": "^4.0.0"
   },
   "devDependencies": {
      "@babel/cli": "7.1.5",
      "@babel/core": "7.1.6",
      "@babel/plugin-proposal-class-properties": "7.1.0",
      "@babel/plugin-proposal-object-rest-spread": "7.0.0",
      "@babel/plugin-transform-runtime": "7.1.0",
      "@babel/preset-env": "7.1.6",
      "@babel/preset-react": "7.0.0",
      "@babel/register": "7.0.0",
      "@babel/runtime": "7.1.5",
      "eslint": "^6.0.0",
      "prettier": "^1.18.2",
      "redux-devtools-extension": "^2.13.7"
   }
}
