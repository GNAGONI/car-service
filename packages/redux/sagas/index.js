import auctions from './auctions';
import auction from './auction';
import contactUs from './contactUs';
import browseByMake from './browseByMake';
import longAuctions from './longAuctions';
import liveAuction from './liveAuction';
import awaitingCollection from './awaitingCollection';
import pendingPayments from './pendingPayments';
import quotesArchive from './quotesArchive';
import carModelsAuction from './carModels';
import user from './user';
import toasts from './toasts';
import quote from './quote';
import scrappedCars from './scrappedCars';

export default [
  ...auctions,
  ...contactUs,
  ...auction,
  ...longAuctions,
  ...liveAuction,
  ...awaitingCollection,
  ...pendingPayments,
  ...quotesArchive,
  ...browseByMake,
  ...carModelsAuction,
  ...user,
  ...toasts,
  ...quote,
  ...scrappedCars,
];
