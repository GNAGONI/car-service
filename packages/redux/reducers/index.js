import { combineReducers } from 'redux';

import auctions from './auctions';
import auction from './auction';
import user from './user';
import userAuth from './userAuth';
import car from './car';
import longAuction from './longAuction';
import dropDown from './dropDown';
import liveAuction from './liveAuction';
import awaitingCollection from './awaitingCollection';
import pendingPayments from './pendingPayments';
import quotesArchive from './quotesArchive';
import modal from './modal';
import browseByMake from './browseByMake';
import location from './location';
import carModels from './carModels';
import message from './message';
import toasts from './toasts';
import quote from './quote';
import scrappedCars from './scrappedCars';

export default combineReducers({
  ...auctions,
  ...auction,
  ...car,
  ...user,
  ...longAuction,
  ...dropDown,
  ...liveAuction,
  ...awaitingCollection,
  ...pendingPayments,
  ...quotesArchive,
  ...modal,
  ...browseByMake,
  ...location,
  ...carModels,
  ...message,
  ...toasts,
  ...quote,
  ...scrappedCars,
  ...userAuth,
});
