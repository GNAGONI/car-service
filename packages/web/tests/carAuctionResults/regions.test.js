import React from 'react'
import renderer from 'react-test-renderer'
import Regions from 'containers/carAuctionResults/regions'
import { connectTestComponentToRedux } from 'test/utils/redux'

it('renders correctly', () => {
  const tree = renderer.create(connectTestComponentToRedux(<Regions />)).toJSON()
  expect(tree).toMatchSnapshot()
})
