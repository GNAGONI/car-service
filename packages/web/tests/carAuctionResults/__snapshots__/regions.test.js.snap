// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="content section-regions bg-before-lighter"
>
  <header
    className="section-head text-center"
  />
  <div
    className="popular-locations text-center"
  >
    <div
      className="list-holder"
    >
      <ul
        className="locations-list"
      />
    </div>
  </div>
  <div
    aria-expanded={null}
    className="personalise-locations text-center collapse"
  >
    <h2>
      Other locations
    </h2>
    <div
      className="tabs-area text-left"
    >
      0
      <div
        className="tab-content"
        id="nav-tabContent"
      />
    </div>
  </div>
  <div
    className="btn-block text-center"
  >
    <button
      className="btn-locations btn btn-primary"
      color="primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div>
        <strong
          className="more"
        >
          View more locations
        </strong>
        <i
          className="icon-chevron-down"
        />
      </div>
    </button>
  </div>
</section>
`;
